### PR TITLE
Implement attributesToRetrieve for getObjects method

### LINKED ISSUE #258: We are not able to specify attributesToRetrieve for getObjects

### PR DESCRIPTION
**Pull request by JanPetr**
Fixes #258

### CODE REVIEW

File: src/AlgoliaSearch/Index.php [hunk 1]
```diff
@@ -210,12 +210,14 @@ public function getObject($objectID, $attributesToRetrieve = null)
...
      * @param array $objectIDs the array of unique identifier of objects to retrieve
+     * @param $attributesToRetrieve (optional) if set, contains the list of attributes to retrieve as a string
+     *                              separated by ","
      *
      * @return mixed
      *
      * @throws \Exception
      */
-    public function getObjects($objectIDs)
+    public function getObjects($objectIDs, $attributesToRetrieve = null)
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: I think we should ask for an array of strings like we do for the other methods.
The documentation is bad for some of the other methods, we say we expect comma separated values where we are actually accepting arrays.

Comment (JanPetr) [inline src/AlgoliaSearch/Index.php]: I wanted to be consistent with [`getObject` method](https://github.com/algolia/algoliasearch-client-php/blob/master/src/AlgoliaSearch/Index.php#L181) where we take comma-separated string.
Definitely not something I'd approve and I'd definitely rather accept `array`, but for me consistency has higher priority right now.

Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: Arg, I just tested and it doesn't work with an array in the 'getObject' method.
I would:
- Make sure we support arrays in both `getObject` and `getObjects`
- Change docbloc and documentation to array
- Keep backward compat

Comment (JanPetr) [inline src/AlgoliaSearch/Index.php]: Good idea, can you review @rayrutjes?

File: src/AlgoliaSearch/Index.php [hunk 2]
```diff
@@ -210,12 +214,14 @@ public function getObject($objectID, $attributesToRetrieve = null)
      * Get several objects from this index.
      *
      * @param array $objectIDs the array of unique identifier of objects to retrieve
+     * @param array $attributesToRetrieve (optional) if set, contains the list of attributes to retrieve as a string
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: I would change the docblock here to 'array of strings' en not mention the 'separated by ,' anymore

File: src/AlgoliaSearch/Index.php [hunk 3]
```diff
@@ -224,6 +230,15 @@ public function getObjects($objectIDs)
         $requests = array();
         foreach ($objectIDs as $object) {
             $req = array('indexName' => $this->indexName, 'objectID' => $object);
+
+            if ($attributesToRetrieve) {
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: This will fail if empty array is provided.

Comment (JanPetr) [inline src/AlgoliaSearch/Index.php]: That's correct behavior. Empty array = all attributes are fetched

Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: It is not the current behaviour. Empty string returns only objectIDs.

Comment (JanPetr) [inline src/AlgoliaSearch/Index.php]: It's the API behavior for `getObject` endpoint. For `getObjects` API returns all attributes on empty string/array.

**Comment by rayrutjes** [comment]
LGTM!

**Pull request merged**